lib/buf.js

```diff
@@ -0,0 +1,30 @@
+const BACK_SLASH = 0x5C;
+
+class InputBuffer {
+    constructor(input) {
+        this._buf = input instanceof Buffer ? input : Buffer.from(input);
```

Comment:
when can input be a Buffer?

Reply:
For now, just for some rigid test cases.